docs/runtime-function-reference.md

```diff
@@ -0,0 +1,691 @@
+# Function Reference
+
+The code runtime built into the server includes a module with functions to implement various logic and custom behavior. It is easy to define authoritative code and conditions on input received by clients.
+
+## Nakama module
+
+This module contains all the core gameplay APIs, all registration functions used at server startup, utilities for various codecs, and cryptographic primitives.
+
+```lua
+local nk = require("nakama")
+```
+
+!!! note
+    All code examples assume the "nakama" module has been imported.
+
+### base16
+
+__base16_decode (input)__
+
+Base 16 decode the input.
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| input | string | The string which will be base16 decoded. |
+
+_Returns_
+
+The base 16 decoded input.
+
+_Example_
+
+```lua
+local decoded = nk.base16_decode("48656C6C6F20776F726C64")
+print(decoded) -- outputs "Hello world"
+```
+
+---
+
+__base16_encode (input)__
+
+Base 16 encode the input.
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| input | string | The string which will be base16 encoded. |
+
+_Returns_
+
+The base 16 encoded input.
+
+_Example_
+
+```lua
+local encoded = nk.base16_encode("Hello world")
+print(encoded) -- outputs "48656C6C6F20776F726C64"
+```
+
+### base64
+
+__base64_decode (input)__
+
+Base 64 decode the input.
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| input | string | The string which will be base64 decoded. |
+
+_Returns_
+
+The base 64 decoded input.
+
+_Example_
+
+```lua
+local decoded = nk.base64_decode("SGVsbG8gd29ybGQ=")
+print(decoded) -- outputs "Hello world"
+```
+
+---
+
+__base64_encode (input)__
+
+Base 64 encode the input.
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| input | string | The string which will be base64 encoded. |
+
+_Returns_
+
+The base 64 encoded input.
+
+_Example_
+
+```lua
+local encoded = nk.base64_encode("Hello world")
+print(encoded) -- outputs "SGVsbG8gd29ybGQ="
+```
+
+### groups
+
+__groups_create (new_groups)__
+
+Setup one or more groups with various configuration settings. The groups will be created if they don't exist or fail if the group names are taken.
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| new_groups | table | The Lua table array of new groups to create. |
+
+_Example_
+
+```lua
+local new_groups = {
+  -- The CreatorId should be a user ID.
+  { Name = "A", Description = "B", Lang = "en", Private = true, CreatorId = "4c2ae592-b2a7-445e-98ec-697694478b1c" }
```

Comment:
Should we mention the available fields that they can populate? like `AvatarUrl`, and `Metadata` ?